File: backend/week5.js
function factorialforloop(nums){
    let result=1
    for(let i=1; i<=nums; i++){
        result=result*i
    }
    return result
}
console.log(factorialforloop(5))

function factorial(n){
    if(n===0 || n===1){
        return 1
    }
    return n*factorial(n-1)

}
console.log(factorial(5))

function isSum(nums, target){
    let seen={}
    for(let i=0; i<nums.length; i++){
        let current=nums[i]
        let need=target-current
        if(need in seen){
            return [seen[need], i]
        }
        seen[current]=i

    }
    return []
}
console.log(isSum([2,3,4,5,8,3], 11))

File: backend/week5.py
def factorial(nums):
    if nums==0 or nums==1:
        return 1
    return nums*factorial(nums-1)
print(factorial(4))

def factorialForloop(n):
    result=1
    for i in range(1, n+1):
        result=result*i
    return result

print(factorialForloop(5))

def isSum(num, target):
    seen={}
    for i in range(len(num)):
        current=num[i]
        need=target-current
        if need in seen:
            return [seen[need], i]
        seen[current]=i
    return []

print(isSum([2,3,4,5,3], 9))